languageservice/src/code-actions/quickfix/add-missing-inputs.ts

```diff
@@ -0,0 +1,67 @@
+import {CodeAction, TextEdit} from "vscode-languageserver-types";
+import {CodeActionProvider} from "../types";
+import {DiagnosticCode, MissingInputsDiagnosticData} from "../../validate-action";
+
+export const addMissingInputsProvider: CodeActionProvider = {
+  diagnosticCodes: [DiagnosticCode.MissingRequiredInputs],
+
+  createCodeAction(context, diagnostic): CodeAction | undefined {
+    const data = diagnostic.data as MissingInputsDiagnosticData | undefined;
+    if (!data) {
+      return undefined;
+    }
+
+    const edits = createInputEdits(data);
+    if (!edits) {
+      return undefined;
+    }
+
+    const inputNames = data.missingInputs.map(i => i.name).join(", ");
+
+    return {
+      title: `Add missing input${data.missingInputs.length > 1 ? "s" : ""}: ${inputNames}`,
+      edit: {
+        changes: {
+          [context.uri]: edits
+        }
+      }
+    };
+  }
+};
+
+function createInputEdits(data: MissingInputsDiagnosticData): TextEdit[] | undefined {
+  const edits: TextEdit[] = [];
+
+  if (data.hasWithKey && data.withIndent !== undefined) {
+    // `with:` exists - use its indentation + 2 for inputs
+    const inputIndent = " ".repeat(data.withIndent + 2);
+
+    const inputLines = data.missingInputs.map(input => {
+      const value = input.default !== undefined ? input.default : '""';
+      return `${inputIndent}${input.name}: ${value}`;
+    });
```

Comment:
I'd probably refacor this, looks the exact same as lines 53-36 https://github.com/actions/languageservices/blob/2d9b787402abd1afe332d694f0eb9ccacb5f357a/languageservice/src/code-actions/quickfix/add-missing-inputs.ts#L53